### PR TITLE
[FLINK-26457] Introduce PartialUpdateMergeFunction

### PR DESCRIPTION
**Pull request by JingsongLi**
PartialUpdateMergeFunction, It will merge two records, completing the not-null fields.
It is very useful for wide tables, where two source tables join together to form a wide table.

### CODE REVIEW

File: docs/content/docs/development/create-table.md [hunk 1]
```diff
@@ -233,3 +233,37 @@ The two methods do not behave in the same way when querying.
...
+
+The value fields are updated to the latest data one by one
+under the same primary key, but null values are not overwritten.
+
+For example, the inputs: 
+- 1, 23.0, 10,   NULL
+- 1, NULL, 20,   'This is a book'
+- 1, 25.2, NULL, NULL
+
+Output: <1, 25.2, 20, 'This is a book'>
```
Comment (LadyForest) [inline docs/content/docs/development/create-table.md:269]: Nit: keep format consistent

For example, the inputs:  
- `<1, 23.0, 10, NULL>`
- `<1, NULL, 20, 'This is a book'>`
- `<1, 25.2, NULL, NULL>`

Output:  
- `<1, 25.2, 20, 'This is a book'>`

File: flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/PartialUpdateMergeFunction.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+public class PartialUpdateMergeFunction implements MergeFunction {
+
+    private final RowData.FieldGetter[] getters;
+
+    public PartialUpdateMergeFunction(RowData.FieldGetter[] getters) {
+        this.getters = getters;
+    }
+
+    private transient GenericRowData row;
```
Comment (LadyForest) [inline flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/PartialUpdateMergeFunction.java:38]: Nit: keep member variables together?

File: docs/content/docs/development/create-table.md [hunk 3]
```diff
@@ -233,3 +233,37 @@ The two methods do not behave in the same way when querying.
...
+  'merge-engine' = 'partial-update'
+);
+```
+
+{{< hint info >}}
+__Note:__ Partial update is only supported for table with primary key.
+{{< /hint >}}
+
+{{< hint info >}}
+__Note:__ Partial update is not supported for streaming consuming.
```
Comment (LadyForest) [inline docs/content/docs/development/create-table.md:258]: Nit: not supported under streaming mode?

**Review by LadyForest — COMMENTED** [review]
Thanks for the contribution! It's almost good to be merged, and I only left some minor comments.

File: docs/content/docs/development/create-table.md (hunk 3 above)
Comment (JingsongLi) [reply docs/content/docs/development/create-table.md:258]: No, just streaming consuming.
User can insert into it under streaming mode.

**Review by LadyForest — APPROVED** [review]
looks good to me :)

**Pull request merged**